test/globalObject/global_object_has_own_property.cc

```diff
@@ -0,0 +1,23 @@
+#include "napi.h"
+
+using namespace Napi;
+
+Value HasPropertyWithCStyleStringAsKey(const CallbackInfo& info) {
+  Object globalObject = info.Env().Global();
+  String key = info[0].As<String>();
+  return Boolean::New(info.Env(),
+                      globalObject.HasOwnProperty(key.Utf8Value().c_str()));
+}
+
+Value HasPropertyWithCppStyleStringAsKey(const CallbackInfo& info) {
+  Object globalObject = info.Env().Global();
+  String key = info[0].As<String>();
+  return Boolean::New(info.Env(), globalObject.HasOwnProperty(key.Utf8Value()));
+}
+
+Value HasPropertyWithNapiValueAsKey(const CallbackInfo& info) {
+  Object globalObject = info.Env().Global();
+  Name key = info[0].As<Name>();
+  return Boolean::New(
+      info.Env(), globalObject.HasOwnProperty(static_cast<napi_value>(key)));
```

Comment:
I think that here it's not necessary to cast the `key` value.  You can do the same like reported below:
```cpp
return Boolean::New(
      info.Env(), globalObject.HasOwnProperty(key));
```